src/backend/parser/gram.y

```diff
@@ -12939,6 +12942,38 @@ GenericType:
 					$$->typmods = $3;
 					$$->location = @1;
 				}
+			| type_function_name attrs opt_type_modifiers WITHOUT TIME ZONE
+				{
+					/*
+					 * This gram rule should only be allowed when we restore
+					 * T-SQL objects from dumped PG SQLs.
+					 *
+					 * Because we have another special handling in printTypmod(),
+					 * this rule can be removed in PG15.
+					 */
+					if (!restore_tsql_datetime2)
+					{
+						ereport(ERROR,
+								(errcode(ERRCODE_SYNTAX_ERROR),
+									errmsg("\"without time zone\" is only valid during restore"),
+									parser_errposition(@4)));
+					}
+					$$ = makeTypeNameFromNameList(lcons(makeString($1), $2));
+					if (strcmp(strVal(linitial($$->names)), "sys") == 0 &&
+						(strcmp(strVal(lsecond($$->names)), "datetime2") == 0 ||
+							strcmp(strVal(lsecond($$->names)), "smalldatetime") == 0))
+					{
+						$$->typmods = $3;
+						$$->location = @1;
+					}
+					else
+					{
+						ereport(ERROR,
+								(errcode(ERRCODE_SYNTAX_ERROR),
```

Comment:
same here, customer should get the exact error they will get without this change

Reply:
fixed